server/battles/BattleResultProcessor.cpp

```diff
@@ -301,53 +301,45 @@ void BattleResultProcessor::endBattle(const CBattleInfoCallback & battle)
 
 void BattleResultProcessor::endBattleConfirm(const CBattleInfoCallback & battle)
 {
-	auto battleQuery = std::dynamic_pointer_cast<CBattleQuery>(gameHandler->queries->topQuery(battle.sideToPlayer(BattleSide::ATTACKER)));
-	if(!battleQuery)
-		battleQuery = std::dynamic_pointer_cast<CBattleQuery>(gameHandler->queries->topQuery(battle.sideToPlayer(BattleSide::DEFENDER)));
-	if(!battleQuery)
-	{
-		logGlobal->trace("No battle query, battle end was confirmed by another player");
-		return;
-	}
-
```

Comment:
Please check replaying autocombat against AI heroes. There were quite a lot of bugs related to those.